### PR TITLE
Add docs on how to build custom client

### PR DESCRIPTION
**Pull request by manuelblum**

### CODE REVIEW

File: docs/docs/how-to-build-custom-client/how-to-build-custom-client.md [hunk 1]
```diff
@@ -0,0 +1,481 @@
+---
+title: How to build custom client
+---
+
+We usually use [next.js](https://nextjs.org/) to build our site client. This guide will show you how to build a custom client for your Comet DXP and render a typical content page from the page tree.
+
+### Page Tree Node List
+
+To load/see all page tree node's, you need to query the `pageTreeNodeList` field in the `Query` type. This field allows you to fetch a list of `PageTreeNode` objects based on specific criteria such as `contentScope` and `category`. Those values differ from project to project and are defined by the project's requirements.
```
Comment (johnnyomair) [inline docs/docs/how-to-build-custom-client/how-to-build-custom-client.md:9]: We also have a paginatedPageTreeNodes query which is more performant -> should we recommend this one?

File: docs/docs/how-to-build-custom-client/how-to-build-custom-client.md [hunk 2]
```diff
@@ -0,0 +1,481 @@
...
+
+With that information in place we can now query the page content, based on the pageTreeNode id.
+
+## Working with the Comet's GraphQL API
+
+The following example demonstrates a Content Page that includes a `HeadlineBlock`, `ImageBlock`, `RichTextBlock`, and a `ColumnBlock` containing two `ImageBlock` elements.
+
+![Sample Content Page](images/sample-content-page.png)
+
+### Loading page data from the API
```
Comment (johnnyomair) [inline docs/docs/how-to-build-custom-client/how-to-build-custom-client.md:55]: I'd strongly suggest mentioning the page tree node by path approach here.

File: docs/docs/how-to-build-custom-client/how-to-build-custom-client.md [hunk 3]
```diff
@@ -0,0 +1,481 @@
...
+    "userGroup": "All"
+}
+```
+
+**HeadlineBlock structure**
+
+-   `key`: A unique identifier for the block.
+-   `visible`: A boolean indicating whether the block is visible.
+-   `type`: The type of the block, in this case, "headline".
+-   `props`: An object containing the block-specific data:
```
Comment (johnnyomair) [inline docs/docs/how-to-build-custom-client/how-to-build-custom-client.md:386]: Those are props of the surrounding blocks block, not the headline block

Comment (manuelblum) [reply docs/docs/how-to-build-custom-client/how-to-build-custom-client.md:386]: Agree, that was the idea, the caption is wrong. Will change this

File: docs/docs/how-to-build-custom-client/how-to-build-custom-client.md (hunk 1 above)
Comment (manuelblum) [reply docs/docs/how-to-build-custom-client/how-to-build-custom-client.md]: i think especially in a tree it's hard to work with paginated stuff. I will add that there is also a more performant query, but i will not suggest one of those.

File: docs/docs/how-to-build-custom-client/how-to-build-custom-client.md (hunk 2 above)
Comment (manuelblum) [reply docs/docs/how-to-build-custom-client/how-to-build-custom-client.md:55]: done.

**Review by thomasdax98 — APPROVED** [review]

**Pull request merged**